app/routes/jokes/new.tsx

```diff
@@ -47,50 +35,24 @@ type ActionData = {
 const badRequest = (data: ActionData) => json(data, { status: 400 });
 
 export const action: ActionFunction = async ({ request }) => {
-  const userId = await requireUserId(request);
-  const form = await request.formData();
-  const name = form.get("name");
-  const content = form.get("content");
-  if (typeof name !== "string" || typeof content !== "string") {
-    return badRequest({
-      formError: `Form not submitted correctly.`,
-    });
-  }
-
-  const fieldErrors = {
-    name: validateJokeName(name),
-    content: validateJokeContent(content),
-  };
-  const fields = { name, content };
-  if (Object.values(fieldErrors).some(Boolean)) {
-    return badRequest({ fieldErrors, fields });
+  const fields = await inputFromForm(request);
+  const result = await createJoke(fields, await getUserId(request));
+  if (!result.success) {
+    return badRequest({ ...result });
   }
-
-  const joke = await db.joke.create({
-    data: { ...fields, jokesterId: userId },
-  });
-  return redirect(`/jokes/${joke.id}`);
+  return redirect(`/jokes/${result.data.id}?redirectTo=/jokes/new`);
 };
 
 export default function NewJokeRoute() {
```

Comment:
minor style nitpicking here, but naming default exports is not really useful. I would use a syntax with an anonymous function such as:
```suggestion
export default function() {
```

or
```suggestion
export default () =>  {
```